packages/wrangler/src/__tests__/deploy.test.ts

```diff
@@ -15905,6 +15910,119 @@ export default{
 
 		expect(deployOutputEntry.autoconfig_summary).toBeUndefined();
 	});
+
+	describe("open-next delegation", () => {
+		async function mockOpenNextLikeProject() {
+			vi.mocked(getInstalledPackageVersion).mockReturnValue("1.14.4");
+
+			fs.mkdirSync("./.open-next/assets", { recursive: true });
+			fs.writeFileSync(
+				"./.open-next/worker.js",
+				"export default { fetch() { return new Response(''); } };"
+			);
+
+			await mockAUSRequest([]);
+
+			writeWorkerSource();
+			mockGetServiceByName("test-name", "production", "dash");
+			writeWranglerConfig(
+				{
+					main: ".open-next/worker.js",
+					compatibility_date: "2024-04-24",
+					compatibility_flags: [
+						"nodejs_compat",
+						"global_fetch_strictly_public",
+					],
+					assets: {
+						binding: "ASSETS",
+						directory: ".open-next/assets",
+					},
+				},
+				"./wrangler.jsonc"
+			);
+			mockSubDomainRequest();
+			mockUploadWorkerRequest({ expectedMainModule: "worker.js" });
+			mockGetServiceBindings("test-name", []);
+			mockGetServiceRoutes("test-name", []);
+			mockGetServiceCustomDomainRecords([]);
+			mockGetServiceSubDomainData("test-name", {
+				enabled: true,
+				previews_enabled: true,
+			});
+			mockGetServiceSchedules("test-name", { schedules: [] });
+			mockGetServiceMetadata("test-name", {
+				created_on: "2025-08-07T09:34:47.846308Z",
+				modified_on: "2025-08-08T10:48:12.688997Z",
+				script: {
+					created_on: "2025-08-07T09:34:47.846308Z",
+					modified_on: "2025-08-08T10:48:12.688997Z",
+					id: "my-worker-id",
+					compatibility_date: "2024-04-24",
+				},
+			} as unknown as ServiceMetadataRes["default_environment"]);
+		}
+
+		it("should delegate to open-next when run in an open-next project and set OPEN_NEXT_DEPLOY", async () => {
+			const runCommandSpy = (await import("../autoconfig/c3-vendor/command"))
+				.runCommand;
+
+			await mockOpenNextLikeProject();
+
+			await runWrangler("deploy");
+
+			expect(runCommandSpy).toHaveBeenCalledOnce();
+			const call = (runCommandSpy as unknown as MockInstance).mock.calls[0];
+			const [command, options] = call;
+			expect(command).toEqual(["npx", "opennextjs-cloudflare", "deploy"]);
+			expect(options).toMatchObject({
+				env: {
+					// Note: we want to ensure that OPEN_NEXT_DEPLOY has been set, this is not strictly necessary but it helps us
+					//       ensure that we can't end up in an infinite wrangler<>open-next invokation loop
+					OPEN_NEXT_DEPLOY: "true",
+				},
+			});
+
+			expect(std.out).toMatchInlineSnapshot(`
+				"
+				 ⛅️ wrangler x.x.x
+				──────────────────
+				The project seems to be an open-next one, calling \`opennextjs-cloudflare deploy\`"
```

Comment:
Up the CLI self confidence :)

```suggestion
				OpenNext project detected, calling \`opennextjs-cloudflare deploy\`"
```